models/measurement.py

```diff
@@ -87,153 +95,147 @@ def subject_join(subject_type):
-                        technique=technique,
-                        value=value,
-                        std=valueStd,
-                        subjectType='strain',
-                        subjectId=strain_id,
-                    ))
+        # TODO (2025-04-03) Figure out how to make a many-to-many relationship
+        metabolites = db_session.scalars(
+            sql.select(Metabolite)
+            .distinct()
+            .join(StudyMetabolite)
+            .where(StudyMetabolite.studyId == study.studyId)
+        ).all()
+
+        for row, time, bioreplicate_uuid, technique in _iterate_over_measurement_csv(db_session, study, csv_string):
+            if technique.subjectType != 'metabolite':
+                continue
+
+            for subject in metabolites:
+                value = row[technique.csv_column_name(subject.name)]
+                if value == '':
+                    value = None
+
+                measurement = Self(
+                    studyId=study.studyId,
+                    bioreplicateUniqueId=bioreplicate_uuid,
+                    position=row['Position'],
+                    timeInSeconds=time,
+                    unit=technique.units,
+                    techniqueId=technique.id,
+                    technique=_generate_technique_name(technique.type, technique.subjectType),
+                    value=value,
+                    subjectId=subject.id,
+                    subjectType='metabolite',
+                )
+                measurements.append(measurement)
 
         db_session.add_all(measurements)
         db_session.commit()
 
         return measurements
+
+
+def _iterate_over_measurement_csv(db_session, study, csv_string):
+    from models import Bioreplicate
+
+    reader = csv.DictReader(StringIO(csv_string), dialect='unix')
+    find_bioreplicate_uuid = functools.cache(Bioreplicate.find_for_study)
+
+    for row in reader:
+        bioreplicate_id = row['Biological_Replicate_id']
+        bioreplicate_uuid = find_bioreplicate_uuid(db_session, study.studyId, bioreplicate_id)
+        time_in_seconds = convert_time(row['Time'], source=study.timeUnits, target='s')
+
+        if bioreplicate_uuid is None:
+            # Missing bioreplicate, skip
+            continue
+
+        for technique in study.measurementTechniques:
+            yield (row, time_in_seconds, bioreplicate_uuid, technique)
+
+
+def _generate_technique_name(technique_type, subject_type):
+    match (technique_type.lower(), subject_type):
+        case ('fc', 'bioreplicate'):
+            return 'FC'
+        case ('fc', 'strain'):
+            return 'FC counts per species'
+        case ('metabolite', _):
+            return 'Metabolites'
+        case ('16s', _):
+            return '16S rRNA-seq'
+        case ('ph', _):
+            return 'pH'
+        case ('od', _):
+            return 'OD'
+        case _:
+            raise ValueError(f"Unknown technique name for type {technique_type}, subject {subject_type}")
```

Comment:
This function provides compatibility with the string "technique" that we have in the database. That way, the export and charts can use that, instead of having to change them right now to use `MeasurementTechnique` objects.